### PR TITLE
Card script for CLB - Aboleth Spawn and modifications to TriggerAbilityTriggered to make it work.

### PR DESCRIPTION
**Pull request by xaviermd**
- Wrote the card script for Aboleth Spawn
- Added 'TriggeredOwnAbility$' to TriggerAbilityTriggered following discussion [here](https://discord.com/channels/267367946135928833/402524622958100501/993591790231896196) regarding the need to provide specific functionality for this card.

### CODE REVIEW

File: forge-gui/res/cardsfolder/upcoming/aboleth_spawn.txt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+Name: Aboleth Spawn
+ManaCost: 2 U
+Types: Creature — Fish Horror
+PT: 2/3
+T:Mode$ AbilityTriggered | TriggerZones$ Battlefield | ValidDestination$ Battlefield | ValidMode$ ChangesZone,ChangesZoneAll | Execute$ DBProbingTelepathy | TriggerDescription$ 6 | ValidSource$ Creature.OppCtrl | TriggeredOwnAbility$ True
+SVar:DBProbingTelepathy:DB$ CopySpellAbility | Defined$ TriggeredSpellAbility | MayChooseTarget$ True | SpellDescription$ Probing Telepathy — Whenever a creature entering the battlefield under an opponent's control causes a triggered ability of that creature to trigger, you may copy that ability. You may choose new targets for the copy.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/aboleth_spawn.txt:6]: Move SpellDescription to TriggerDescription

File: forge-gui/res/cardsfolder/upcoming/aboleth_spawn.txt [hunk 2]
```diff
@@ -0,0 +1,7 @@
+Name: Aboleth Spawn
+ManaCost: 2 U
+Types: Creature — Fish Horror
+PT: 2/3
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/aboleth_spawn.txt:4]: Keywords missing & Formatting

File: forge-gui/res/cardsfolder/upcoming/aboleth_spawn.txt [hunk 3]
```diff
@@ -0,0 +1,7 @@
+Name: Aboleth Spawn
+ManaCost: 2 U
+Types: Creature — Fish Horror
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/aboleth_spawn.txt:3]: Remove — from types, handled by GUI

File: forge-game/src/main/java/forge/game/trigger/TriggerAbilityTriggered.java [hunk 4]
```diff
@@ -87,6 +87,20 @@ public final boolean performTest(final Map<AbilityKey, Object> runParams) {
                 return false;
             }
         }
+        
+        if (hasParam("TriggeredOwnAbility") && "True".equals(getParam("TriggeredOwnAbility")))
+        {
+            boolean match = false;
```
Comment (Hanmac) [inline forge-game/src/main/java/forge/game/trigger/TriggerAbilityTriggered.java:93]: shouldn't that be better with `Iterables.contains(causes, source)`?

also the part above with "ValidCause" can be shorted to just `matchesValidParam("ValidCause", causes)`

it is smart enough to handle iterables

**Review by tool4ever — APPROVED** [review]

**Comment by xaviermd** [comment]
Closes #809

**Pull request merged**